File: app/src/main/java/torille/fi/lurkforreddit/comments/CommentComponent.java
package torille.fi.lurkforreddit.comments;

import dagger.Component;
import torille.fi.lurkforreddit.data.RedditRepositoryComponent;
import torille.fi.lurkforreddit.di.scope.FragmentScoped;

/**
 * Created by eva on 24.4.2017.
 */
@FragmentScoped
@Component(dependencies = RedditRepositoryComponent.class, modules = {CommentPresenterModule.class})
public interface CommentComponent {

    void inject(CommentFragment fragment);

}

File: app/src/main/java/torille/fi/lurkforreddit/comments/CommentComponent.kt
package torille.fi.lurkforreddit.comments

import dagger.Component
import torille.fi.lurkforreddit.data.RedditRepositoryComponent
import torille.fi.lurkforreddit.di.scope.FragmentScoped

/**
 * Dagger 2 component that provides [RedditRepositoryComponent]
 */
@FragmentScoped
@Component(dependencies = arrayOf(RedditRepositoryComponent::class), modules = arrayOf(CommentPresenterModule::class))
interface CommentComponent {

    fun inject(fragment: CommentFragment)

}
